editors/code/src/ctx.ts

```diff
@@ -233,6 +233,19 @@ export class Ctx implements RustAnalyzerExtensionApi {
                     this.outputChannel!.show();
                 }),
             );
+            this.pushClientCleanup(
+                this._client.onNotification(ra.unindexedProject, async (params) => {
+                    const command = `${this.config.discoverProjectRunner}.discoverWorkspaceCommand`;
```

Comment:
Does this work when `discoverProjectRunner` is `undefined` or will executing then throw?

Reply:
You're right, this would absolutely panic/throw. I _assumed_ that if notifications are configured, a `discoverProjectRunner` would also be configured, but I don't think we should. I'll make this check if `discoverProjectRunner` is set, but arguably, one should imply the other.

Do you think it'd be a good idea to error out before initializing the server here? I don't think the client notifications are useful without a mechanism to _discover_ other projects.

Reply:
Seems fine, as long as we don't have the throwing behavior here

Reply:
Whoops, sorry—performance stuff was happening this past week. fixed.